packages/spark/components/masthead.js

```diff
@@ -39,37 +88,63 @@ const hideMobileNavs = () => {
   document.documentElement.classList.remove('sprk-u-Overflow--hidden');
   document.body.classList.remove('sprk-u-Height--100');
   document.documentElement.classList.remove('sprk-u-Height--100');
-  getElements('[data-sprk-mobile-nav]', (item) => {
+  getElements('[data-sprk-mobile-nav]', item => {
     item.classList.add('sprk-u-Display--none');
   });
-  getElements('.sprk-c-Menu__icon--open', (item) => {
+  getElements('.sprk-c-Menu__icon--open', item => {
     item.classList.remove('sprk-c-Menu__icon--open');
   });
-  getElements('.sprk-c-Masthead--open', (item) => {
+  getElements('.sprk-c-Masthead--open', item => {
     item.classList.remove('sprk-c-Masthead--open');
   });
 };
 
-const hideSelectorMask = (mastheadSelectorMask) => {
+const hideSelectorMask = mastheadSelectorMask => {
   mastheadSelectorMask.classList.remove('sprk-c-MastheadMask');
 };
 
-const showSelectorMask = (mastheadSelectorMask) => {
+const showSelectorMask = mastheadSelectorMask => {
   mastheadSelectorMask.classList.add('sprk-c-MastheadMask');
 };
 
 const bindUIEvents = () => {
-  getElements('[data-sprk-mobile-nav-trigger]', (element) => {
+  getElements('[data-sprk-mobile-nav-trigger]', element => {
     const mainLayout = document.querySelector('[data-sprk-main]');
     const masthead = document.querySelector('[data-sprk-masthead]');
-    const selectorDropdown = document.querySelector('[data-sprk-dropdown="dropdown-selector"]');
-    const selectorTrigger = document.querySelector('[data-sprk-dropdown-trigger="dropdown-selector"]');
+    const selectorDropdown = document.querySelector(
+      '[data-sprk-dropdown="dropdown-selector"]',
+    );
+    const selectorTrigger = document.querySelector(
+      '[data-sprk-dropdown-trigger="dropdown-selector"]',
+    );
 
     const nav = document.querySelector(
-      `[data-sprk-mobile-nav="${element.getAttribute('data-sprk-mobile-nav-trigger')}"]`,
+      `[data-sprk-mobile-nav="${element.getAttribute(
+        'data-sprk-mobile-nav-trigger',
+      )}"]`,
     );
+    /*
+    * Check if the mobile menu is visible
+    * on page and set scroll event
+    */
+    let isMenuVisible;
+    window.addEventListener('load', () => {
+      isMenuVisible = isElementVisible('.sprk-c-Masthead__menu');
+      toggleScrollEvent(isMenuVisible);
+    });
+    /*
+    * If the mobile menu visibility changes
+    * toggle scroll event listener
+    */
+    window.addEventListener('resize', () => {
```

Comment:
@afebbraro @RCopeland  it looks like both the highlight board and stepper are also listening for resize. We should at least debounce these methods if not think about consolidating them so we don't spam the resize callback API

Reply:
That sounds good, I'm all for consolidating!